### PR TITLE
Add IVT for UnitTestingHostReloadService

### PR DESCRIPTION
**Pull request by vritant24**
1. Add IVT to use the UnitTestingHostReloadService
2. Update API to give a list of capabilities
3. Update API to dictate whether or not to commit updates

### CODE REVIEW

**Comment by vritant24** [comment]
tagging @tmat

File: src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingHotReloadService.cs [hunk 1]
```diff
@@ -65,23 +69,25 @@ public UnitTestingHotReloadService(HostWorkspaceServices services)
         /// </summary>
         /// <param name="solution">Solution that represents sources that match the built binaries on disk.</param>
         /// <param name="cancellationToken">Cancellation token.</param>
-        public async Task StartSessionAsync(Solution solution, CancellationToken cancellationToken)
+        public async Task StartSessionAsync(Solution solution, ImmutableArray<string> _capabilities, CancellationToken cancellationToken)
```
Comment (tmat) [inline src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingHotReloadService.cs:72]: >_capabilities [](http://example.com/codeflow?start=86&length=13)

Style: no `_`

File: src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingHotReloadService.cs [hunk 2]
```diff
@@ -65,23 +69,25 @@ public UnitTestingHotReloadService(HostWorkspaceServices services)
...
+        /// for the first invocation.
         /// </summary>
         /// <param name="solution">Solution snapshot.</param>
+        /// <param name="shouldCommit">commits changes if true, discards if false</param>
         /// <param name="cancellationToken">Cancellation token.</param>
         /// <returns>
         /// Updates (one for each changed project) and Rude Edit diagnostics. Does not include syntax or semantic diagnostics.
         /// </returns>
-        public async Task<(ImmutableArray<Update> updates, ImmutableArray<Diagnostic> diagnostics)> EmitSolutionUpdateAsync(Solution solution, CancellationToken cancellationToken)
+        public async Task<(ImmutableArray<Update> updates, ImmutableArray<Diagnostic> diagnostics)> EmitSolutionUpdateAsync(Solution solution, bool shouldCommit, CancellationToken cancellationToken)
```
Comment (tmat) [inline src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingHotReloadService.cs:90]: >shouldCommit [](http://example.com/codeflow?start=148&length=12)

Better name: `commitUpdates`

**Comment by vritant24** [comment]
Done. Ready for merge

File: src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingHotReloadService.cs [hunk 3]
```diff
@@ -19,17 +19,21 @@ internal sealed class UnitTestingHotReloadService
     {
         private sealed class DebuggerService : IManagedEditAndContinueDebuggerService
         {
-            public static readonly DebuggerService Instance = new();
+            private readonly ImmutableArray<string> _capabilities;
+
+            public DebuggerService(ImmutableArray<string> _capabilities)
+            {
+                this._capabilities = _capabilities;
+            }
```
Comment (davidwengier) [inline src/Features/Core/Portable/ExternalAccess/UnitTesting/API/UnitTestingHotReloadService.cs:27]:
```suggestion
            public DebuggerService(ImmutableArray<string> capabilities)
            {
                _capabilities = capabilities;
            }
```

**Review by davidwengier — APPROVED** [review]
LGTM. Will merge once the build is green.

**Review by tmat — APPROVED** [review]

**Comment by vritant24** [comment]
@davidwengier looks like the ci build failed due to an agent availability issue? Is there a way to re-run it?

**Comment by tmat** [comment]
Done

**Pull request merged**